pkg/reconciler/vspheresource/vsphere.go

```diff
@@ -146,18 +146,17 @@ func (r *Reconciler) reconcileConfigMap(ctx context.Context, vms *sourcesv1alpha
 
 func (r *Reconciler) reconcileServiceAccount(ctx context.Context, vms *sourcesv1alpha1.VSphereSource) error {
 	ns := vms.Namespace
-	name := resourcenames.ServiceAccount(vms)
-
-	_, err := r.saLister.ServiceAccounts(ns).Get(name)
-	if apierrs.IsNotFound(err) {
+	serviceAccountName := vms.Spec.ServiceAccountName
+	_, err := r.saLister.ServiceAccounts(ns).Get(serviceAccountName)
+	if apierrs.IsNotFound(err) && serviceAccountName == "default" {
 		sa := resources.MakeServiceAccount(ctx, vms)
 		_, err := r.kubeclient.CoreV1().ServiceAccounts(ns).Create(ctx, sa, metav1.CreateOptions{})
```

Comment:
Looks like we're still creating the service account if it doesn't exist, did you mean to resolve the previous comment?

Reply:
If the default service account does not exist whatsoever, this is a good failsafe for that

Reply:
The default service accounts are created automatically when namespaces are created. 

I feel it's a UX issue. Since this was prompted by poor UX when accessing a private registry:

A scenario where the user forgets to create the service account, with the current code, it will create an empty service account. The pod won't be able to start due to lack of privileges. It might be better to show a `service account not found` to the user rather than create an empty version of what should've been created by them.

Reply:
Kk! Convinced me (maybe we should change that on the horizon source too, I'll create the issue)